src/fromager/overrides.py

```diff
@@ -83,6 +83,15 @@ def patches_for_source_dir(
     return sorted((patches_dir / source_dir_name).glob("*.patch"))
 
 
+def patches_for_req(patches_dir: pathlib.Path, req_name: str) -> list[pathlib.Path]:
+    """
+    This function will return patches for a specific requirement.
+    It takes in patches directory and requirement as input.
+    The output will be a list of all directories containing patches for that requirement
+    """
+    return sorted((patches_dir).glob(f"{req_name}*"))
```

Comment:
We have some packages that start with a common prefix, so this is going to return more than it should. For example, we have `instructlab` and `instructlab-sdg`. Any patches for `instructlab-sdg` will appear to be related to `instructlab`. The directory naming convention includes a `-` between the package name and version, so you could add that to the pattern to fix this.

There will also be patches in a directory just named for the requirement, without any version. For that case, you could just check for that directory to exist at all and include it if it is there without using a glob pattern.

The `patches_for_source_dir()` function takes the source directory, instead of a requirement, because the source directory doesn't have to match the requirement name. Requirements like `instructlab.sdg` and `instructlab-sdg` are equivalent. It looks like you're expecting the prefix of the source directory name, without any version information. This function might be clearer if it took the full source directory name and did its own parsing of that name to strip the version string, instead of having the caller do that step.